src/_pytest/assertion/rewrite.py

```diff
@@ -126,7 +127,7 @@ def find_spec(
         ):
             return None
         else:
-            fn = spec.origin
+            self.fn = fn = spec.origin
```

Comment:
i see no reason add this random state - the spec is used to get it anyway

Reply:
This is needed inside `get_code` as the argument is the `fullname` of the module rather than the file path. The same thing is done in the `FileLoader` from the stdlib https://github.com/python/cpython/blob/2433cc79d79d9c1db8e53d4b9bde26e9a47fb0b9/Lib/importlib/_bootstrap_external.py#L924

Reply:
as the object you changed is not just a loader, but also a MetaPathFinder - it would be necessary to split those objects before the state would be permissible as there would be more than one loader instance with different values

Reply:
cpython has those split up as well - see https://github.com/python/cpython/blob/2433cc79d79d9c1db8e53d4b9bde26e9a47fb0b9/Lib/importlib/_bootstrap_external.py#L1267 for reference

Reply:
> it would be necessary to split those objects before the state would be permissible

I could try to create a full name -> origin mapping on the finder instead, to keep the changes small. How does that sound?